Disallow remotes beginning with dash in a backwards compatible way

#### What is the problem you are having with rclone?
Can't use remotes beginning with the character `-`.
 
#### What is your rclone version (output from `rclone version`)
```
rclone v1.51.0
- os/arch: linux/amd64
- go version: go1.13.7
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Fedora 32


####  Which cloud storage system are you using? (eg Google Drive)
Any


#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
`rclone lsf -my-remote-:` or `rclone lsf "-my-remote-:"`


#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
Error: unknown shorthand flag: 'm' in -my-remote-:
Usage:
  rclone lsf remote:path [flags]

Flags:
      --absolute           Put a leading / in front of path names.
      --csv                Output in CSV format.
  -d, --dir-slash          Append a slash to directory names. (default true)
      --dirs-only          Only list directories.
      --files-only         Only list files.
  -F, --format string      Output format - see  help for details (default "p")
      --hash h             Use this hash when h is used in the format MD5|SHA-1|DropboxHash (default "MD5")
  -h, --help               help for lsf
  -R, --recursive          Recurse into the listing.
  -s, --separator string   Separator for the items in the format. (default ";")

Use "rclone [command] --help" for more information about a command.
Use "rclone help flags" for to see the global flags.
Use "rclone help backends" for a list of supported services.

2020/05/20 22:46:47 Fatal error: unknown shorthand flag: 'm' in -my-remote-:
```

contributors.md: `go get -u github.com/rclone/rclone`
`cmd/serve/ftp/ftp.go:154:3: unknown field 'PublicIp' in struct literal of type server.ServerOpts (but does have PublicIP)`
It shows up while testing:
```
$ go test -v
# github.com/rclone/rclone/cmd/serve/ftp
cmd/serve/ftp/ftp.go:154:3: unknown field 'PublicIp' in struct literal of type server.ServerOpts (but does have PublicIP)
```
Though the # makes me think it's commented out, but go tries to get it an

all atempts to `go run`, `go build` `go run rclone.go` `go run rclone` get me
```
# github.com/rclone/rclone/cmd/serve/ftp
cmd/serve/ftp/ftp.go:154:3: unknown field 'PublicIp' in struct literal of type server.ServerOpts (but does have PublicIP)
```

--- a/fs/fspath/path.go
+++ b/fs/fspath/path.go
@@ -19,6 +19,7 @@ const (
 var (
 	errInvalidCharacters = errors.New("config name contains invalid characters - may only contain 0-9, A-Z ,a-z ,_ , - and space ")
 	errCantBeEmpty       = errors.New("can't use empty string as a path")
+	errCantStartWithDash = errors.New("config name starts with -")
 
 	// urlMatcher is a pattern to match an rclone URL
 	// note that this matches invalid remoteNames
@@ -35,6 +36,10 @@ var (
 func CheckConfigName(configName string) error {
 	if !configNameMatcher.MatchString(configName) {
 		return errInvalidCharacters
+	}
+	// Reject configName, if it starts with -, complicates usage. (#4261)
+	if strings.HasPrefix(configName, "-") {
+		return errCantStartWithDash
 	}
 	return nil
 }

--- a/fs/fspath/path_test.go
+++ b/fs/fspath/path_test.go
@@ -23,6 +23,9 @@ func TestCheckConfigName(t *testing.T) {
 		{"rem\\ote", errInvalidCharacters},
 		{"[remote", errInvalidCharacters},
 		{"*", errInvalidCharacters},
+		{"-remote", errCantStartWithDash},
+		{"r-emote-", nil},
+		{"_rem_ote_", nil},
 	} {
 		got := CheckConfigName(test.in)
 		assert.Equal(t, test.want, got, test.in)

